### PR TITLE
Pte 19 add game menu

### PR DESCRIPTION
**Pull request by Misiek22113**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
-# This is a sample Python script.
+import pygame
+from menu.Menu import Menu
+from pygame.locals import *
```
Comment (Hidennnn) [inline main.py]: Import shouldn't be done with ' * '. Enumerate imports.

File: main.py [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-    # Use a breakpoint in the code line below to debug your script.
-    print(f'Hi, {name}')  # Press Ctrl+F8 to toggle the breakpoint.
+menu.handle_main_menu_loop()
 
 
-# Press the green button in the gutter to run the script.
-if __name__ == '__main__':
-    print_hi('PyCharm')
 
-# See PyCharm help at https://www.jetbrains.com/help/pycharm/
```
Comment (Hidennnn) [inline main.py:16]: Too many blank lines

File: menu/Menu.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+from menu.window.window import Window
+
+
+class Menu(Window):
+
+    def __init__(self, width, height, name="Menu"):
+        super().__init__(name, width, height)
+        self.width = width
+        self.height = height
+        self.PLAY_BUTTON = None
```
Comment (Hidennnn) [inline menu/Menu.py]: You can do static like "EMPTY_BUTTON"

File: menu/Menu.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        self.CONTROLS_BUTTON = None
+        self.EXIT_BUTTON = None
+        self.MENU_MOUSE_POS = None
+
+    def print_window_menu(self):
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
+        self.draw_text("PASS THE EXAM", 640, 100, self.FONT_HEADER, self.TEXT_COLOR)
+        self.PLAY_BUTTON = Button(image=self.MENU_BUTTON, pos=(640, 350), text_input="PLAY", font=self.FONT_OPTION,
```
Comment (Hidennnn) [inline menu/Menu.py]: This and another tuples can be static like "FIRST_LINE", "SECOND_LINE"

File: menu/Menu.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        self.width = width
+        self.height = height
+        self.PLAY_BUTTON = None
+        self.CONTROLS_BUTTON = None
+        self.EXIT_BUTTON = None
+        self.MENU_MOUSE_POS = None
+
+    def print_window_menu(self):
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
+        self.draw_text("PASS THE EXAM", 640, 100, self.FONT_HEADER, self.TEXT_COLOR)
```
Comment (Hidennnn) [inline menu/Menu.py]: is "640, 100" position?

File: menu/Menu.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        super().__init__(name, width, height)
+        self.width = width
+        self.height = height
+        self.PLAY_BUTTON = None
+        self.CONTROLS_BUTTON = None
+        self.EXIT_BUTTON = None
+        self.MENU_MOUSE_POS = None
+
+    def print_window_menu(self):
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
```
Comment (Hidennnn) [inline menu/Menu.py]: some static?

File: menu/Menu.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+            for button in [self.PLAY_BUTTON, self.CONTROLS_BUTTON, self.EXIT_BUTTON]:
+                button.change_color(self.MENU_MOUSE_POS)
+                button.update(self.screen)
+
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    pygame.quit()
+                    sys.exit()
+                if event.type == pygame.MOUSEBUTTONDOWN:
```
Comment (Hidennnn) [inline menu/Menu.py]: What's you think about enter before this?

File: menu/MenuController/MenuController.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+from menu.Menu import Menu
+from menu.controls.Controls import Controls
+from menu.lobby.Lobby import Lobby
+from menu.champion_select.champion_select import ChampionSelect
+
+
+class MenuController:
+
+    def __init__(self):
+        self.menu = Menu(1280, 720)
```
Comment (Hidennnn) [inline menu/MenuController/MenuController.py]: 1280, 720 - SCREEN_WIDTH, SCREEN_HEIGHT - add static

File: menu/MenuController/MenuController.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+        while True:
+            if actual == "menu":
+                actual = self.menu.handle_menu_loop()
+            elif actual == "controls":
+                actual = self.controls.handle_controls_loop()
+            elif actual == "play" or actual == "champion_select":
+                actual = self.champion_select.handle_champion_select_loop()
+            elif actual == "back":
+                actual = self.menu.handle_menu_loop()
```
Comment (Hidennnn) [inline menu/MenuController/MenuController.py]: The same like actual == "menu"

File: menu/button/Button.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    def __init__(self, image, pos, text_input, font, base_color, hovering_color):
+        self.image = image
+        self.x_pos = pos[0]
+        self.y_pos = pos[1]
+        self.font = font
+        self.base_color = base_color
+        self.hovering_color = hovering_color
+        self.text_input = text_input
+        self.text = self.font.render(self.text_input, True, self.base_color)
+        if self.image is None:
```
Comment (Hidennnn) [inline menu/button/Button.py]: if not self.image

File: menu/button/Button.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            self.image = self.text
+        self.rect = self.image.get_rect(center=(self.x_pos, self.y_pos))
+        self.text_rect = self.text.get_rect(center=(self.x_pos+4, self.y_pos+4))
```
Comment (Hidennnn) [inline menu/button/Button.py]: "4" is magic number - make static

File: menu/button/Button.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+    def update(self, screen):
+        if self.image is not None:
```
Comment (Hidennnn) [inline menu/button/Button.py]: if self.image

File: menu/button/Button.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+        screen.blit(self.text, self.text_rect)
+
+    def check_for_input(self, position):
+        if position[0] in range(self.rect.left, self.rect.right) and position[1] in range(self.rect.top,
+                                                                                          self.rect.bottom):
+            return True
+        return False
+
+    def change_color(self, position):
+        if position[0] in range(self.rect.left, self.rect.right) and position[1] in range(self.rect.top,
```
Comment (Hidennnn) [inline menu/button/Button.py]: if position[0] in range(self.rect.left, self.rect.right) \
                and position[1] in range(self.rect.top, self.rect.bottom):
                
 also check if self.rect.left < position[0] < self.rect.right ect. works

File: menu/button/Button.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+        self.rect = self.image.get_rect(center=(self.x_pos, self.y_pos))
+        self.text_rect = self.text.get_rect(center=(self.x_pos+4, self.y_pos+4))
+
+    def update(self, screen):
+        if self.image is not None:
+            screen.blit(self.image, self.rect)
+        screen.blit(self.text, self.text_rect)
+
+    def check_for_input(self, position):
+        if position[0] in range(self.rect.left, self.rect.right) and position[1] in range(self.rect.top,
```
Comment (Hidennnn) [inline menu/button/Button.py]: if position[0] in range(self.rect.left, self.rect.right) \
and position[1] in range(self.rect.top, self.rect.bottom):

File: menu/champion_select/champion_select.py [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+        self.BUTTON_ARROW_LEFT = None
+        self.width = width
+        self.height = height
+        self.BACK_BUTTON = None
+        self.NEXT_BUTTON = None
+        self.CHAMPION_SELECT_MOUSE_POS = None
+
+    def print_champion_select_menu(self):
+        scale = 10
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: What is (0, 0)?

File: menu/champion_select/champion_select.py [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        self.BUTTON_ARROW_RIGHT = None
+        self.BUTTON_ARROW_LEFT = None
+        self.width = width
+        self.height = height
+        self.BACK_BUTTON = None
+        self.NEXT_BUTTON = None
+        self.CHAMPION_SELECT_MOUSE_POS = None
+
+    def print_champion_select_menu(self):
+        scale = 10
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: maybe 10 as static

File: menu/champion_select/champion_select.py [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+        self.draw_text("Choose your Champion", 640, 100, self.FONT_CHAMPION_SELECT, self.TEXT_COLOR)
+        self.draw_text(self.CHAMPIONS[self.champion_index][1], 640, 500, self.FONT_CHAMPION_SELECT, self.TEXT_COLOR)
+        self.BACK_BUTTON = Button(image=self.MENU_BUTTON, pos=(440, 650), text_input="BACK", font=self.FONT_OPTION,
+                                  base_color=self.BASE_COLOR, hovering_color=self.HOVERING_COLOR)
+        self.NEXT_BUTTON = Button(image=self.MENU_BUTTON, pos=(840, 650), text_input="NEXT", font=self.FONT_OPTION,
+                                  base_color=self.BASE_COLOR, hovering_color=self.HOVERING_COLOR)
+        self.BUTTON_ARROW_LEFT = Button(image=None, pos=(400, 350), text_input="<", font=self.FONT_OPTION,
+                                        base_color=self.BASE_COLOR, hovering_color=self.HOVERING_COLOR)
+        self.BUTTON_ARROW_RIGHT = Button(image=None, pos=(880, 350), text_input=">", font=self.FONT_OPTION,
+                                         base_color=self.BASE_COLOR, hovering_color=self.HOVERING_COLOR)
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: maybe change tuples in pos to static with name of position on screen

File: menu/champion_select/champion_select.py [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+        self.width = width
+        self.height = height
+        self.BACK_BUTTON = None
+        self.NEXT_BUTTON = None
+        self.CHAMPION_SELECT_MOUSE_POS = None
+
+    def print_champion_select_menu(self):
+        scale = 10
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
+        self.draw_text("Choose your Champion", 640, 100, self.FONT_CHAMPION_SELECT, self.TEXT_COLOR)
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: what is 640, 100?

File: menu/champion_select/champion_select.py [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+        elif to_change < 0:
+            self.champion_index = max_index
+
+    def handle_champion_select_loop(self):
+        while True:
+            self.print_champion_select_menu()
+
+            self.CHAMPION_SELECT_MOUSE_POS = pygame.mouse.get_pos()
+
+            for button in self.BUTTON_ARROW_LEFT, self.BUTTON_ARROW_RIGHT, self.BACK_BUTTON, self.NEXT_BUTTON:
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: Do variables after "in" as list - more consistent

File: menu/champion_select/champion_select.py [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+                if event.type == pygame.MOUSEBUTTONDOWN:
+                    if self.BACK_BUTTON.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        return "back"
+                    if self.NEXT_BUTTON.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        return "lobby"
+                    if self.BUTTON_ARROW_RIGHT.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        self.change_champion_index(self.champion_index + 1)
+                    if self.BUTTON_ARROW_LEFT.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        self.change_champion_index(self.champion_index - 1)
+                if event.type == pygame.QUIT:
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: Take pygame.QUIT on top - more consistent

File: menu/champion_select/champion_select.py [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+            for event in pygame.event.get():
+                if event.type == pygame.MOUSEBUTTONDOWN:
+                    if self.BACK_BUTTON.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        return "back"
+                    if self.NEXT_BUTTON.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        return "lobby"
+                    if self.BUTTON_ARROW_RIGHT.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        self.change_champion_index(self.champion_index + 1)
+                    if self.BUTTON_ARROW_LEFT.check_for_input(self.CHAMPION_SELECT_MOUSE_POS):
+                        self.change_champion_index(self.champion_index - 1)
```
Comment (Hidennnn) [inline menu/champion_select/champion_select.py]: 1s can be constant

File: menu/controls/Controls.py [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+        self.width = width
+        self.height = height
+        self.BACK_BUTTON = None
+        self.EXIT_BUTTON = None
+        self.CONTROLS_MOUSE_POS = pygame.mouse.get_pos()
+        self.KEY_W = pygame.image.load("assets/controls/key_w.png")
+        self.KEY_S = pygame.image.load("assets/controls/key_s.png")
+        self.KEY_A = pygame.image.load("assets/controls/key_a.png")
+        self.KEY_D = pygame.image.load("assets/controls/key_d.png")
+        self.KEY_W = pygame.transform.scale(self.KEY_W,
```
Comment (Hidennnn) [inline menu/controls/Controls.py]: All arguments of function can be in one line (below)

File: menu/controls/Controls.py [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+
+    def print_controls(self):
+        key_rect = self.KEY_W.get_rect(center=(640, 450))
+        self.screen.blit(self.KEY_W, key_rect)
+        key_rect = self.KEY_A.get_rect(center=(490, 600))
+        self.screen.blit(self.KEY_A, key_rect)
+        key_rect = self.KEY_S.get_rect(center=(640, 600))
+        self.screen.blit(self.KEY_S, key_rect)
+        key_rect = self.KEY_D.get_rect(center=(790, 600))
+        self.screen.blit(self.KEY_D, key_rect)
```
Comment (Hidennnn) [inline menu/controls/Controls.py]: tuples with positions can be constants

File: menu/controls/Controls.py [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+
+    def print_controls_menu(self):
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
+        self.print_controls()
+        self.draw_text("CONTROLS", 640, 100, self.FONT_HEADER, self.TEXT_COLOR)
+        self.BACK_BUTTON = Button(image=self.MENU_BUTTON, pos=(640, 650), text_input="BACK", font=self.FONT_OPTION,
+                                  base_color=self.BASE_COLOR, hovering_color=self.HOVERING_COLOR)
```
Comment (Hidennnn) [inline menu/controls/Controls.py]: Add constant

File: menu/lobby/Lobby.py [hunk 25]
```diff
@@ -0,0 +1,51 @@
+import pygame, sys
```
Comment (Hidennnn) [inline menu/lobby/Lobby.py]: one import for one line

File: menu/lobby/Lobby.py [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+        self.LOBBY_MOUSE_POS = None
+
+    def print_lobby_menu(self):
+        self.screen.blit(self.BACKGROUND_IMAGE, (0, 0))
+        self.draw_text("Waiting for other players...", 640, 100, self.FONT_OPTION, self.TEXT_COLOR)
+        self.draw_text("Player 1", 240, 200, self.FONT_OPTION, self.TEXT_COLOR)
+        self.draw_text("Player 2", 640, 200, self.FONT_OPTION, self.TEXT_COLOR)
+        self.draw_text("Player 3", 1040, 200, self.FONT_OPTION, self.TEXT_COLOR)
+        self.LEAVE_BUTTON = Button(image=self.MENU_BUTTON, pos=(640, 650), text_input="LEAVE", font=self.FONT_OPTION,
+                                   base_color=self.BASE_COLOR, hovering_color=self.HOVERING_COLOR)
```
Comment (Hidennnn) [inline menu/lobby/Lobby.py]: Add constants

File: menu/Menu.py (hunk 5 above)
Comment (Misiek22113) [reply menu/Menu.py]: it is

File: menu/Menu.py (hunk 4 above)
Comment (Misiek22113) [reply menu/Menu.py]: for now I'll leave it how it is because this coordinates will be changed

File: menu/Menu.py (hunk 7 above)
Comment (Misiek22113) [reply menu/Menu.py]: to be honest i dont really know where you want to use enter

File: menu/MenuController/MenuController.py (hunk 9 above)
Comment (Misiek22113) [reply menu/MenuController/MenuController.py]: yea I'll move that to line 19

File: menu/button/Button.py (hunk 11 above)
Comment (Misiek22113) [reply menu/button/Button.py]: ok

File: menu/button/Button.py (hunk 14 above)
Comment (Misiek22113) [reply menu/button/Button.py]: you mean I should split this line to two before and?

File: menu/champion_select/champion_select.py (hunk 15 above)
Comment (Misiek22113) [reply menu/champion_select/champion_select.py]: It has to be like that to cover whole window with image

File: menu/champion_select/champion_select.py (hunk 18 above)
Comment (Misiek22113) [reply menu/champion_select/champion_select.py]: position of button

File: menu/Menu.py (hunk 7 above)
Comment (Hidennnn) [reply menu/Menu.py]: after 42nd line

Comment (Misiek22113) [reply menu/Menu.py]: yea it should be there

**Review by Hidennnn — APPROVED** [review]

**Pull request merged**